manifests/interface.pp

```diff
@@ -16,28 +16,60 @@
 # @param config_dir
 #   Path to wireguard configuration files
 define wireguard::interface (
-  Variant[Array,String] $address,
-  String                $private_key,
-  Integer[1,65535]      $listen_port,
-  Enum['present','absent'] $ensure = 'present',
+  Variant[Array,String]     $address,
+  Variant[String, Deferred] $private_key,
```

Comment:
This would drop support for all Puppet version > 6 because the Type "Deferred" isn't known by Puppet 4/5.

I've implemented this without knowing someone else already did it. I think I found solution which supports Puppet 4/5/6.